frontend/src/tabs-hook.ts

```diff
@@ -47,18 +47,24 @@ class TabsHook extends Logger {
     const self = this;
     const tree = (document.getElementById('root') as any)._reactRootContainer._internalRoot.current;
     let scrollRoot: any;
-    let currentNode = tree;
+    async function findScrollRoot(currentNode: any, iters: number): Promise<any> {
+      if (iters >= 30) {
+        return null;
```

Comment:
This is where we would sleep and retry instead of returning null.

Reply:
I have made the changes you requested in a new commit, please review when you have a moment.